README.md

```diff
@@ -22,7 +23,116 @@ Or install it yourself as:
 
 ## Usage
 
-[API documentation](http://www.rubydoc.info/gems/factree)
+*For details, check out the [API documentation](http://www.rubydoc.info/gems/factree).*
+
+### Finding paths through a decision function
+
+Factree provides tools for making choices based on a set of facts that are not yet known. You write a decision function that takes a set facts and returns a conclusion. Factree will run your function and make sure it has all of the facts it needs to complete. If any are missing, Factree will tell you what's needed to continue.
+
+For example, say I want to pick an animal based on its attributes. First I'll write a function to make the decision.
+
+```ruby
+decide = ->(facts) do
```

Comment:
Does `find_path` use `call`? If so you can add some examples which specify other callable types such as method references and objects that implement .call like

```ruby
class IsHerbivore
  def self.call
  end
end
```

When you can do with Factree.find_path IsHerbivore